js/ui/text_box/ui.text_editor.base.js

```diff
@@ -347,11 +350,23 @@ const TextEditorBase = Editor.inherit({
     _collapseInputContainer: function() {
         const $element = this.$element();
         const isIE11 = browser.msie && browser.version <= 11;
-        if(isIE11 && $element.css('display') === 'block') {
+        const parentElement = $element.parent();
+
+        if(isIE11 && $element.css('display') === 'block' && (!window.getComputedStyle(parentElement.get(0)) || parentElement.innerWidth() > this._editorMinWidth())) {
             $element.addClass(TEXTEDITOR_COLLAPSED_CLASS);
         }
     },
 
+    _editorMinWidth: function() {
+        const $input = this._input();
+        const buttonsWidth = ($(this._$beforeButtonsContainer).width() || 0) +
```

Comment:
Do we need a default value for the `width()` function result (`|| 0`)?

Reply:
Yes, the result of $(this._$beforeButtonsContainer).width() can be calculated as undefined

Reply:
Ok. I suggest extracting intermediate results to corresponding constants.